rust/feature-flags/src/flag_matching.rs

```diff
@@ -805,6 +827,41 @@ impl FeatureFlagMatcher {
         }
     }
 
+    /// Evaluates dynamic cohort property filters
+    ///
+    /// NB: This method first caches all of the cohorts associated with the team, which allows us to avoid
+    /// hitting the database for each cohort filter.
+    pub async fn evaluate_cohort_filters(
+        &self,
+        cohort_property_filters: &[PropertyFilter],
+        target_properties: &HashMap<String, Value>,
+    ) -> Result<bool, FlagError> {
+        // Caching all of the cohorts like this will make it so that we don't have to hit the database for each cohort filter
+        let cohort_cache =
```

Comment:
As per the other PR - these should be cached at a higher level than the per-request basis, and use something like `moka` to maintain  TTL etc. As far as I can tell, the current code hits PG per-request for:
- Person properties
- Teams cohorts
- Each static cohort individually

You really want to only be hitting PG per-request for the person properties (and considering "is this person in this static cohort" as a property of the person), everything else should be cached across-requests (being careful, ofc, to not run OOM - have every item in the cache be ~the same size, set a hard upper limit on cached item count).